sylvia-derive/src/strip_self_path.rs

```diff
@@ -13,3 +15,29 @@ impl Fold for StripSelfPath {
         syn::fold::fold_path(self, Path { segments, ..path })
     }
 }
+
+pub struct AddSelfPath<'a, GenericT>(&'a [&'a GenericT]);
```

Comment:
This is something I'd rather avoid, but as for now I still don't have enouch knowledge about sylvia. I'm happy to once again talk about adding Self to generics and maybe have some brainstorming on other solutions.

Reply:
Agree. We can maybe setup some biweekly or something calls and discuss stuff like that there